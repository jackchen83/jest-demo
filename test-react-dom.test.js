// __tests__/CheckboxWithLabel-test.js

import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import CheckboxWithLabel from './CheckboxWithLabel';

test('snapshot render', () => {
  const checkbox = renderer.create(<CheckboxWithLabel labelOn="On" labelOff="Off" />);

  expect(checkbox.toJSON()).toMatchSnapshot();
  checkbox.unmount();
});

it('CheckboxWithLabel changes the text after click', () => {
  // Render a checkbox with label in the document
  const checkbox = mount(<CheckboxWithLabel labelOn="On" labelOff="Off" />);

  expect(checkbox.text()).toEqual('Off');

  checkbox.find('input').simulate('change');

  expect(checkbox.text()).toEqual('On');

  expect(checkbox).toMatchSnapshot();
  checkbox.unmount();

});

