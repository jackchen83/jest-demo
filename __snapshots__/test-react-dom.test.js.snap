// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel changes the text after click 1`] = `
<CheckboxWithLabel
  labelOff="Off"
  labelOn="On"
>
  <label>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    On
  </label>
</CheckboxWithLabel>
`;

exports[`snapshot render 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Off
</label>
`;
